### ISSUE TITLE
Call to randn not inlined in randn!

### ISSUE THREAD

**Issue by antoine-levitt** [JuliaLang/julia#31607]
From https://discourse.julialang.org/t/possible-inefficiency-in-randn/22732/: 

```
using Random
using BenchmarkTools

broadcast_randn(x) = (x .= randn.())
function myrandn!(x)
    @inbounds for i in eachindex(x)
        x[i] = randn()
    end
    x
end

x = zeros(10_000)
@btime broadcast_randn($x)
@btime myrandn!($x)
@btime randn!($x)
```
is slow in the last one. This appears to be because the call to `randn` is not inlined, despite it being marked as `@inline`.

Also, `randn` is marked `@inline` but `randexp`, which is similar, is not. Either both or none should be marked `@inline`.

EDIT:
```
julia> @btime broadcast_randn($x);
  32.109 μs (0 allocations: 0 bytes)

julia> @btime myrandn!($x);
  32.007 μs (0 allocations: 0 bytes)

julia> @btime randn!($x)
  52.698 μs (0 allocations: 0 bytes)
```

**Comment by KristofferC** [comment]
Looks fixed?

```
julia> @btime broadcast_randn($x);
  47.669 μs (0 allocations: 0 bytes)

julia> @btime myrandn!($x);
  48.105 μs (0 allocations: 0 bytes)

julia> @btime randn!($x);
  38.506 μs (0 allocations: 0 bytes)
```

**Comment by mbauman** [comment]
Now actually fixed in 1.7-DEV (all times in μs with 0 allocations: 0 bytes):

|  | `@btime broadcast_randn($x)` | `@btime myrandn!($x)` | ` @btime randn!($x)` |
|-|-|-|-|
| **1.0** | 26.884 | 27.414 | 54.064 |
| **1.2** | 27.369 | 26.405 | 54.827 |
| **1.4** | 43.034 | 43.064 | 56.143 |
| **1.5** | 48.968 | 48.976 | 23.651 |
| **1.6** | 41.278 | 42.793 | 20.058 |
| **1.7-DEV** | 27.250 | 27.370 | 17.857 |
